src/NUnitEngine/nunit.engine.core.tests/Drivers/NUnitFrameworkDriverTests.cs

```diff
@@ -13,22 +12,39 @@
 namespace NUnit.Engine.Drivers
 {
     // Functional tests of the NUnitFrameworkDriver calling into the framework.
-    public class NUnit3FrameworkDriverTests
+#if NETFRAMEWORK
+    [TestFixture("2009")]
+    [TestFixture("2018")]
+#endif
+    public class NUnitFrameworkDriverTests
     {
         private const string MOCK_ASSEMBLY = "mock-assembly.dll";
-        private const string LOAD_MESSAGE = "Method called without calling Load first";
+        private const string LOAD_MESSAGE = "Method called without calling Load first. Possible error in runner.";
 
         private IDictionary<string, object> _settings = new Dictionary<string, object>();
 
-        private NUnit3FrameworkDriver _driver;
+        private NUnitFrameworkDriver _driver;
         private string _mockAssemblyPath;
 
+#if NETFRAMEWORK
+        private string _whichApi;
+        public NUnitFrameworkDriverTests(string whichApi)
+        {
+            _whichApi = whichApi;
+        }
+#endif
```

Comment:
If you only put the `2009` Test case inside `#if NETFRAMEWORK` you can make this code unconditional.
This then allows you to move `RunTestsAction_WithInvalidFilterElement_ThrowsException` outside `#if FRAMEWORK` so that you also test invalid elements for .NET Core.

Reply:
Nice!